src/include/distributed/multi_physical_planner.h

```diff
@@ -182,15 +182,23 @@ typedef struct Task
 	uint32 taskId;
 
 	/*
-	 * If queryString != NULL, then we have a single query for all placements.
-	 * Otherwise, length of perPlacementQueryStrings is equal to length of
-	 * taskPlacementList and can assign a different query for each placement.
-	 * We need this flexibility when a query should return node specific values.
-	 * For example, on which node did we succeed storing some result files?
+	 * For most queries either query or queryStringLazy is not NULL. This means
+	 * we have a single query for all placements.
+	 * If this is not the case, the length of perPlacementQueryStrings is
+	 * non-zero and equal to length of taskPlacementList. Like this it can
+	 * assign a different query for each placement. We need this flexibility
+	 * when a query should return node specific values. For example, on which
+	 * node did we succeed storing some result files?
+	 *
+	 * In almost all cases queryStringLazy should be read by using
+	 * TaskQueryString(). This will populate the field if only the query field
+	 * is not NULL.
 	 */
-	char *queryString;
+	Query *query;
```

Comment:
Why do we need this in addition to jobQuery + relationShardList?

Reply:
The main reason is that we don't have access to the job in all places where we currently use `task->query`.

Reply:
Can we change that? This blows up the data structure quite a bit.

Reply:
as discussed in person this is not possible or at least not easy in some cases. So this will be kept like it is

Reply:
I'd still like a better name and comment.

Reply:
Done, now called `queryForLocalExecution`